scalyr_agent/builtin_monitors/syslog_utils.py

```diff
@@ -0,0 +1,211 @@
+            match_hash = ""
+        full_path = os.path.join(
+            self._global_config.agent_log_path,
+            file_template.safe_substitute({"HASH": match_hash}),
+        )
+        matched_config = JsonObject({"parser": "syslog", "path": full_path})
+        matched_config.update(config)
+
+        def config_matcher(unit):
+            if regex.match(unit) is not None:
+                return matched_config
+            return None
+
+        return config_matcher
+
+    def get_logger(self, unit):
+        result = LogConfigManager.get_logger(self, unit)
+        self.log_deleter.add_current_log(self.get_config(unit)["path"])
+        return result
+
+    def check_for_old_logs(self):
+        self.log_deleter.check_for_old_logs()
+
+
+class LogDeleter(object):
+    """Deletes unused log files that match a log_file_template"""
+
+    def __init__(
+        self,
+        check_interval_mins,
+        delete_interval_hours,
+        check_rotated_timestamps,
+        max_log_rotations,
+        log_file_template,
+        agent_log_path,
+    ):
+        self._check_interval = check_interval_mins * 60
+        self._delete_interval = delete_interval_hours * 60 * 60
+        self._check_rotated_timestamps = check_rotated_timestamps
+        self._max_log_rotations = max_log_rotations
+        self._log_glob = os.path.join(
+            agent_log_path, log_file_template.safe_substitute(HASH="*")
+        )
+
+        self._last_check = time.time()
+        self.existing_logs = []
+
+    def add_current_log(self, path):
+        self.existing_logs.append(path)
+
+    def _get_old_logs_for_glob(
+        self, current_time, glob_pattern, existing_logs, check_rotated, max_rotations
+    ):
+
+        result = []
+
+        for matching_file in glob.glob(glob_pattern):
+            try:
+                added = False
+                mtime = os.path.getmtime(matching_file)
+                if (
+                    current_time - mtime > self._delete_interval
+                    and matching_file not in existing_logs
+                ):
+                    result.append(matching_file)
+                    added = True
+
+                for i in range(max_rotations, 0, -1):
+                    rotated_file = matching_file + (".%d" % i)
+                    try:
+                        if not os.path.isfile(rotated_file):
+                            continue
+
+                        if check_rotated:
+                            mtime = os.path.getmtime(rotated_file)
+                            if current_time - mtime > self._delete_interval:
+                                result.append(rotated_file)
+                        else:
+                            if added:
+                                result.append(rotated_file)
+
+                    except OSError as e:
+                        global_log.warn(
+                            "Unable to read modification time for file '%s', %s"
```

Comment:
If we are only catching this specific exception, it would make sense to only wrap that line where we can except that exception (aka ``os.path.getmtime``).